### PR TITLE
dependency: bump go.opentelemetry.io/otel from v1.14.0 to v1.15.0

### PR DESCRIPTION
**Pull request by chaochn47**
Noticed by dependabot https://github.com/etcd-io/etcd/pull/15806

Related discussion https://github.com/etcd-io/etcd/pull/15776#issuecomment-1530534819

Please read https://github.com/etcd-io/etcd/blob/main/CONTRIBUTING.md#contribution-flow.

### CODE REVIEW

**Comment by chaochn47** [comment]
TLDR, this version bump is a incompatible change. We should ignore this dependabot notification until the gRPC [issue](https://github.com/etcd-io/etcd/blob/main/Documentation/contributor-guide/dependency_management.md#about-grpc) is sorted out. 

The following is the explanation of etcd build failures

```
make build 

stderr: # go.opentelemetry.io/otel/exporters/otlp/otlptrace/internal/otlpconfig
stderr: /home/chaochn/.gvm/pkgsets/go1.19.7/global/pkg/mod/go.opentelemetry.io/otel/exporters/otlp/otlptrace@v1.11.2/internal/otlpconfig/options.go:100:62: undefined: internal.GetUserAgentHeader

FAIL: (code:2):
  % (cd server && 'env' 'CGO_ENABLED=0' 'GO_BUILD_FLAGS= -v -mod=readonly' 'GOOS=linux' 'GOARCH=amd64' 'go' 'build' '-v' '-mod=readonly' '-trimpath' '-installsuffix=cgo' '-ldflags=-X=go.etcd.io/etcd/api/v3/version.GitSHA=6b002478c' '-o=../bin/etcd' '.')
FAIL: etcd_build (GOARCH=amd64)
make: *** [build] Error 2
```

`otlptracegrpc` will be stuck in `v1.11.2` because its direct dependency on `google.golang.org/grpc v1.51.0`

https://github.com/etcd-io/etcd/blob/57206f9dfb2740e42c9ae229e0c8411e49e8ab0a/server/embed/config_tracing.go#L49-L53

when initializing the exporter, the `internal.GetUserAgentHeader()` is only defined in `v1.14.0`. 

```
// New constructs a new Exporter and starts it.
func New(ctx context.Context, opts ...Option) (*otlptrace.Exporter, error) {
	return otlptrace.New(ctx, NewClient(opts...))
}
```

```
func newClient(opts ...Option) *client {
	cfg := otlpconfig.NewGRPCConfig(asGRPCOptions(opts)...)
```

```
import "go.opentelemetry.io/otel/exporters/otlp/internal"

// NewGRPCConfig returns a new Config with all settings applied from opts and
// any unset setting using the default gRPC config values.
func NewGRPCConfig(opts ...GRPCOption) Config {
	cfg := Config{
		Traces: SignalConfig{
			Endpoint:    fmt.Sprintf("%s:%d", DefaultCollectorHost, DefaultCollectorGRPCPort),
			URLPath:     DefaultTracesPath,
			Compression: NoCompression,
			Timeout:     DefaultTimeout,
		},
		RetryConfig: retry.DefaultConfig,
		DialOptions: []grpc.DialOption{grpc.WithUserAgent(internal.GetUserAgentHeader())},
	}
```

**Comment by chaochn47** [comment]
cc @ahrtr I will close the PR and corresponding dependabot one if there is no objection. 

I wanted to add this information to [dependency management doc](https://github.com/etcd-io/etcd/blob/main/Documentation/contributor-guide/dependency_management.md#about-grpc) but I think it's better to spend time on resolving https://github.com/etcd-io/etcd/issues/15145 firstly.

**Comment by ahrtr** [comment]
> cc @ahrtr I will close the PR and corresponding dependabot one if there is no objection.


I have no objection. Thank you for the confirmation. 

> I wanted to add this information to [dependency management doc](https://github.com/etcd-io/etcd/blob/main/Documentation/contributor-guide/dependency_management.md#about-grpc)

Please feel free to deliver a PR. Thanks
